keps/sig-scheduling/4832-async-preemption/README.md

```diff
@@ -0,0 +1,999 @@
+bogged down.
+-->
+
+#### Story 1
+
+We have a plugin running the preemption at PostFilter,
+but we don't want to make any API calls at PostFilter because it slows down the whole scheduling throughput.
+
+After this KEP is implemented, we determine which Pod(s) to preempt at `PostFilter`,
+nominate the Pod based on the calculation,
+and actually makes the API calls in the goroutine.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+#### When kube-apiserver is unstable
+
+When kube-apiserver is unstable and API calls at the preemption goroutine fails frequently,
+the scheduler could make a non-best scheduling result 
+because the scheduler nominates pods at `PostFilter` though, those Pods won't be scheduled on nodes because the preemption API calls fail.
+
+Let's say many mid-priority Pods are making the preemption API calls.
+During the preemption goroutine for them are runnning, the scheduler assumes they'll be scheduled at the Nodes eventually
+that the preemptions are targeting via `.Status.NominatedNodeName`.
+So, other mid-priority or lower priority Pods' scheduling take those preempter Pods into consideration, 
+which is correct if the preemption goroutine finishes successful actually, while which results in non-best scheduling results.
+(Higher priority Pods won't be affected; Pods can take place of reserved for lower priority Pods via `.Status.NominatedNodeName`)
+
+But, in the first place though, when kube-apiserver is unstable, the scheduler doesn't behave well 
+because it works with a lot of communication with kube-apiserver.
+Even if the scheduler makes the best scheduling result, the binding API might fail after all.
+
+So, we don't have to pay a special attention to this issue.
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+To achieve an asynchronous preemption, we will change the preemption plugin's implementation like the following:
+1. The preemption PostFilter plugin calculates the preemption target and nominate the Pod for the Node. (We'll use `AddNominatedPod` API exposed from the scheduling framework to plugins.)
```

Comment:
this is just an in-memory operation, right?

What about the API call to set the nominatedNodeName?

Reply:
API call to set the nominated node name will be done within the preemption goroutine after all victim Pods are deleted successfully. Will mention it.